docker/README.md

```diff
@@ -91,72 +73,159 @@ once and saved. (This is done in CentOS by using the already existing
+In order to save the necessary dependencies, call `./docker/run_dockers.bsh` on 
+`git-lfs-full-build_*.dockerfile` before `git-lfs_*.dockerfile` and the rpms 
+will be extracted from the bootstrapped images and saved in the `./repo` directory.
+(This _can_ be done in one command. Order matters)
+
+    ./docker/run_dockers.bsh git-lfs-full-build_*.dockerfile git-lfs_*.dockerfile
 
 This is most important for CentOS 6 where git 1.8.2 or newer is not available, 
 only git 1.7.1 is available, so every user either has to build git from source, 
-or use the rpms generated by the `git-lfs-full-build_centos_6` image. This 
-will only needs to be done once.
+or use the rpms generated by the `git-lfs-full-build_centos_6` image. Calling
+the environment building images only needs to be done once, they should remain in
+the `./repo` directory afterwards.
 
-(To manually build a docker image, run 
-`docker build -t $(basename ${DockerDir}) -f ${DockerDir}/Dockerfile ./docker`
+### Run Docker Environment Variables ###
 
+There are a few environment variables you can set to easily adjust the behavior
+of the `run_docker.bsh` script.
+
+`export` before calling `run_docker.bsh`
+
+`REPO_HOSTNAME` - Override the hostname for all the repos generated/tested (see below)
+
+`DOCKER_AUTOBUILD` - Default 1. `run_docker.bsh` always calls `build_docker.bsh`
+to ensure your docker image is up-to-date. Sometimes you may not want this. If 
+set to 0, it will not build docker images before running.
+
+`AUTO_REMOVE` - Default 1. Docker containers are automatically deleted on 
+exit. If set to 0, the docker containers will not be automatically deleted upon 
+exit. This can be useful for a post mortem analysis (using other docker commands
+not covered here). Just make sure you clean up the docker containers manually.
+
+### Build Docker Environment Variables ###
+
+`export` before calling `run_docker.bsh`/`build_docker.bsh`. 
+
+`DOCKER_LFS_BUILD_VERSION` - The version of LFS used to bootstrap the (CentOS)
+environment. This does not need to be bumped every version. This can be a tag 
+or a sha.
 
 ##Deploying/Building Repositories##
 
-When `./docker/run_dockers.bsh` is done building git-lfs and the rpms/deb,
-it actually creates a repository for distribution too. Each distro gets a repo
-generated in `./repos/{DISTRO_NAME}/{VERSION #}`. Just drop the repo
+When `./docker/run_dockers.bsh` is done building git-lfs and generating packages,
+it automatically creates a repository for distribution too. Each distro gets a
+repo generated in `./repos/{DISTRO_NAME}/{VERSION #}`. Just drop the repo
 directory onto a webserver and you have a fully functioning Linux repo. (See
 Testing the Repositories below for more detail)
 
 The two major packages included are:
-git-lfs-....* - the git-lfs package
-git-lfs-repo-release....* - A package to install the repo.
+`git-lfs-....*` - the git-lfs package
+`git-lfs-repo-release....*` - A package to install the repo.
```

Comment:
`git-lfs-....*` is the actual git-lfs package containing the git-lfs executable
`git-lfs-repo-release....*` is a package that will install a repo described by `REPO_HOSTNAME` that when hosted, can provide the user with git-lfs through their package manager program (yum, apt-get, etc...). If the user downloads `git-lfs-repo-release....*` they do not need to download `git-lfs-....*`